goSrc/src/vmCaller/iroha/commands.go

```diff
@@ -423,6 +423,176 @@ func GetRolePermissions(role string) ([]pb.RolePermission, error) {
+		if err!=nil || err1!=nil {
+			return TxPaginationMeta, fmt.Errorf("Invalid lastTxTime value")
+		}else{
+			TxPaginationMeta.OptLastTxTime = &pb.TxPaginationMeta_LastTxTime{lastTime}
+		}
+	}
+	// check firstTxHeight
+	if len(firstTxHeight) != 0 {
+		firstHeightInt, err := strconv.ParseUint(firstTxHeight, 10, 64)
+		if err!=nil {
+			return TxPaginationMeta, fmt.Errorf("Invalid firstTxHeight value")
+		}else{
+			TxPaginationMeta.OptFirstTxHeight = &pb.TxPaginationMeta_FirstTxHeight{firstHeightInt}
+		}
+	}
+	// check lastTxHeight
+	if len(lastTxHeight) != 0 {
+		lastHeightInt, err := strconv.ParseUint(lastTxHeight, 10, 64)
+		if err!=nil {
+			return TxPaginationMeta, fmt.Errorf("Invalid lastTxHeight value")
+		}else{
+			TxPaginationMeta.OptLastTxHeight = &pb.TxPaginationMeta_LastTxHeight{lastHeightInt}
+		}
+	}
+	return TxPaginationMeta, nil
+}
+func GetAccountTransactions(accountID string, pageSize string, firstTxHash string, ordering string, firstTxTime string, lastTxTime string, firstTxHeight string, lastTxHeight string) ([]*pb.Transaction, error) {
+	txPagination, err := makeTxPaginationMeta(pageSize, ordering, firstTxTime, lastTxTime, firstTxHeight, lastTxHeight)
+	if err != nil {
+		return []*pb.Transaction{}, err
+	}
+	query := &pb.Query{Payload: &pb.Query_Payload{
+		Meta: &pb.QueryPayloadMeta{
+			CreatedTime:      uint64(time.Now().UnixNano() / int64(time.Millisecond)),
+			CreatorAccountId: Caller,
+			QueryCounter:     1},
+		Query: &pb.Query_Payload_GetAccountTransactions{
+			GetAccountTransactions: &pb.GetAccountTransactions{AccountId: accountID,
+															   PaginationMeta: &txPagination}}}}
+	queryResponse, err := makeProtobufQueryAndExecute(IrohaQueryExecutor, query)
+	if err != nil {
+		return []*pb.Transaction{}, err
+	}
+	switch response := queryResponse.Response.(type) {
+	case *pb.QueryResponse_ErrorResponse:
+		return []*pb.Transaction{}, fmt.Errorf(
+			"ErrorResponse in GetAccountTransactions: %d, %v",
+			response.ErrorResponse.ErrorCode,
+			response.ErrorResponse.Message,
+		)
+	case *pb.QueryResponse_TransactionsPageResponse:
+		transactionsPageResponse := queryResponse.GetTransactionsPageResponse()
+		return transactionsPageResponse.Transactions, nil
+	default:
+		return []*pb.Transaction{}, fmt.Errorf("Wrong response type in GetAccountTransactions")
+	}
+}
+
+func GetPendingTransactions(pageSize string, firstTxHash string, ordering string, firstTxTime string, lastTxTime string, firstTxHeight string, lastTxHeight string) ([]*pb.Transaction, error) {
+	txPagination, err := makeTxPaginationMeta(pageSize, ordering, firstTxTime, lastTxTime, firstTxHeight, lastTxHeight)
+	if err != nil {
+		return []*pb.Transaction{}, err
+	}
+	query := &pb.Query{Payload: &pb.Query_Payload{
+		Meta: &pb.QueryPayloadMeta{
+			CreatedTime:      uint64(time.Now().UnixNano() / int64(time.Millisecond)),
+			CreatorAccountId: Caller,
+			QueryCounter:     1},
```

Comment:
I see the part of code multiple times, can't it be extracted to separate method?